data_safe_haven/infrastructure/programs/sre/data.py

```diff
@@ -425,6 +426,47 @@ def __init__(
                 resource_group_name=kwargs["resource_group_name"],
             )
         )
+        # Add diagnostic setting for files
+        insights.DiagnosticSetting(
+            f"{storage_account_data_configuration._name}_diagnostic_setting",
+            name=f"{storage_account_data_configuration._name}_diagnostic_setting",
+            log_analytics_destination_type="Dedicated",
+            logs=[
+                {
+                    "category_group": "allLogs",
+                    "enabled": True,
+                    "retention_policy": {
+                        "days": 0,
+                        "enabled": False,
+                    },
+                },
+                {
+                    "category_group": "audit",
+                    "enabled": True,
+                    "retention_policy": {
+                        "days": 0,
+                        "enabled": False,
+                    },
+                },
+            ],
+            metrics=[
+                {
+                    "category": "Transaction",
+                    "enabled": True,
+                    "retention_policy": {
+                        "days": 0,
+                        "enabled": False,
+                    },
+                }
+            ],
+            resource_uri=storage_account_data_configuration.id.apply(
+                # This is the URI of the fileService resource which is automatically
+                # created.
+                lambda resource_id: resource_id
+                + "/fileServices/default"
+            ),
```

Comment:
```suggestion
            # This is the URI of the automatically created fileService resource
            resource_uri=Output.concat(storage_account_data_configuration.id, "/fileServices/default"),
```

Reply:
I was certain there was a helper method to do this but couldn't find it! 😆